Dockerfile

```diff
@@ -1,5 +1,11 @@
 # Base image to use for the final stage
 ARG base_image=public.ecr.aws/amazonlinux/amazonlinux:2
+FROM alpine:3.15 as alpine
```

Comment:
In ACK, we use images from `public ecr` because of docker pull throttling

Reply:
Thanks, really good point. And nice to know public ECR mirrors docker hub, great job! docker-rate-limit is a pain.
Should base_image ARG also be removed from Dockerfile, Makefile, CI, etc since it's no longer used? Or leave that for a separate PR?
And yeah, definitely worth testing locally with a few controllers 😄

Reply:
1. I was unable to find legitimate/popular `scratch` image on `public ecr` :/ , so we'll probably continue to use `amazonlinux` version until a `scratch` image is published on `public ecr` . I did find that new release of `amazonlinux(public.ecr.aws/amazonlinux/amazonlinux:2018.03.0.20211222.0)` does not have security vulnerabilities so we will probably use that before migrating to `scratch`
2. Given the fact that ACK team will enable regular scanning on controller images, if this `amazonlinux` image needs further update, ACK team will be made aware of it.
3. I will try out your non-root solution with `amazonlinux` locally and comment on PR if it worked.

Thank you for your amazing contribution.
(Re: Golang vulnerabilities, ACK needs upstream controller runtime updates but there have been changes in upstream API(`Object` parameter type), which will need some refactoring in ACK runtime. I am looking into it.)

Reply:
Thanks :) Obviously your call, but scratch is a special container startpoint which contains nothing, don't think it actually counts as a pull unless running a really old version of Docker (it's literally an empty container, no FS, no shell, nothing, all that's in there is your go binary). bit more info here: https://hub.docker.com/_/scratch. It's quite popular among containers running go binaries since go produces a single self-contained executable including the runtime

Reply:
I will try to publish `scratch` image into ACK's `public-ecr-repo` and hopefully we will be able to use it instead of docker hub `scratch`. I will update here if I will successfully achieve it. :)

Reply:
excellent point! testing locally now with your changes as is.